CSharp/ArticleProjects/DotNetTutorial/Program.cs

```diff
@@ -286,17 +286,33 @@ private static string GetContainerSasUrl(CloudBlobClient blobClient, string cont
         /// <param name="resourceFiles">A collection of <see cref="ResourceFile"/> objects representing blobs within
         /// a Storage account container. The StartTask will download these files from Storage prior to execution.</param>
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
-        private static async Task CreatePoolAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
+        private static async Task CreatePoolIfNotExistAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
         {
-            Console.WriteLine("Creating pool [{0}]...", poolId);
-
-            // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
-            // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
-            CloudPool pool = batchClient.PoolOperations.CreatePool(
-                poolId: poolId,
-                targetDedicated: 3,                                                         // 3 compute nodes
-                virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
-                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2
+            CloudPool pool = null;
+            try
+            {
+                Console.WriteLine("Creating pool [{0}]...", poolId);
+
+                // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
+                // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
+                pool = batchClient.PoolOperations.CreatePool(
+                    poolId: poolId,
+                    targetDedicated: 3,                                                         // 3 compute nodes
+                    virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
+                    cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2

```

Comment:
Hi Ivan -- somehow I missed that you had already responded to this.

I definitely do want `await pool.CommitAsync` up in the try block. The `CreatePool` call just creates the pool object in memory, the `CommitAsync` bit is what actually submits it to the service, which means that the `CommitAsync` bit is what will throw the `PoolExists` error. If the `CommitAsync` isn't in the try, then the catch will never be hit. Also, the `StartTask` is part of the pool, so it makes sense to not edit it unless you're creating a new pool (because if the pool already exists it presumably also already has the start task).